lib/Parse/Lexer.cpp

```diff
@@ -622,6 +622,13 @@ static bool isLeftBound(const char *tokBegin, const char *bufferBegin) {
   // The first character in the file is not left-bound.
   if (tokBegin == bufferBegin) return false;
 
+  if (tokBegin - 1 != bufferBegin) {
+    if (tokBegin[-2] == '*' && tokBegin[-1] == '/') {
+      // End of a slash-star comment, which we treat as whitespace.
+      return false;
+    }
+  }
+
```

Comment:
OK, I've moved it to a separate case, which I think reads better than in the default case since it also depends on `tokBegin[-1]`.